File: src/main/java/com/example/airmed/Entity/PsychiatricData.java
package com.example.airmed.Entity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class PsychiatricData {
    @Id
    @GeneratedValue(strategy =  GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, updatable = false)
    private Long id;

    @Column(nullable = false)
    private boolean presant;

    @Column(nullable = false)
    private String diagnostics;

    @Column(columnDefinition = "boolean default false")
    private boolean hospitalization;

    @Column
    private String antidepressant;

    @Column
    private String moodStabilizers;

    @Column
    private String antipsychotics;

    @Column
    private String suicideThoughts;

    @ManyToOne
    @JoinColumn(name = "patient", nullable = false)
    private Patient patient;
}

File: frontend/src/classes/PsychiatricData.ts
import {Patient} from "./Patient.ts";

export class PsychiatricData {
    id: number;
    presant: boolean;
    diagnostics: string;
    hospitalization: boolean;
    antidepressant: string;
    moodStabilizers: string;
    antipsychotics: string;
    suicideThoughts: string;
    patient: Patient;


    constructor(id: number, presant: boolean, diagnostics: string, hospitalization: boolean, antidepressant: string, moodStabilizers: string, antipsychotics: string, suicedeThoughts: string, patient: Patient) {
        this.id = id;
        this.presant = presant;
        this.diagnostics = diagnostics;
        this.hospitalization = hospitalization;
        this.antidepressant = antidepressant;
        this.moodStabilizers = moodStabilizers;
        this.antipsychotics = antipsychotics;
        this.suicideThoughts = suicedeThoughts;

        this.patient = new Patient(patient.id,patient.pnc,patient.firstName,
            patient.lastName,patient.mail, patient.phone, patient.psychiatrist,
            patient.psychotherapist
        );
    }
}

export function convertToPsychiatricData(data: any): PsychiatricData {
    return new PsychiatricData(
        data.id,
        data.presant,
        data.diagnostics,
        data.hospitalization,
        data.antidepressant,
        data.moodStabilizers,
        data.antipsychotics,
        data.suicedeThoughts,
        new Patient(
            data.patient.id,
            data.patient.pnc,
            data.patient.firstName,
            data.patient.lastName,
            data.patient.mail,
            data.patient.phone,
            data.patient.psychiatrist,
            data.patient.psychotherapist
        )
    );
}
